packages/libs/pactjs-generator/src/contract/parsing/utils/parser-utilities/inspect.ts

```diff
@@ -0,0 +1,30 @@
+import { wrapData } from '../dataWrapper';
+import { IPointer } from '../getPointer';
+import { ExceptKeywords } from '../typeUtilities';
+
+import { FAILED, IParser, rule, RuleReturn, RuleReturnType } from './rule';
+
+interface IInspector {
+  /**
+   *
+   * @param name name can be any string, except for "inspect," which is a reserved name.
+   */
+  <T extends string, P extends IParser>(
+    name: ExceptKeywords<T, 'inspect'>,
+    parser: P,
+  ): (pointer: IPointer) => RuleReturn<RuleReturnType<P>, T>;
+  <P extends IParser>(parser: P): (
+    pointer: IPointer,
+  ) => RuleReturn<RuleReturnType<P>, undefined>;
+}
+
+export const $: IInspector = (one: string | IParser, second?: IParser) =>
+  rule((pointer) => {
+    const name = typeof one === 'string' ? one : undefined;
+    if (name === 'inspect') throw new Error('inspect is a reserved name');
+    const parser = second || (one as IParser);
+    const result = parser(pointer);
+    if (result === FAILED) return FAILED;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return wrapData(result, name) as any;
```

Comment:
You could use `unknown` here, that would satisfy the rule, and as it's recast by the interface it'll be fine for the caller

```suggestion
    return wrapData(result, name) as unknown;
```

Maybe this is a solution for other places as well